account-kit/react/src/components/auth/modal.tsx

```diff
@@ -16,19 +16,20 @@ export const AuthModal = () => {
     })
   );
 
-  const { setAuthStep } = useAuthContext();
+  const { setAuthStep, authStep } = useAuthContext();
   const { isOpen, closeAuthModal, openAuthModal } = useAuthModal();
 
   const handleSignup = useCallback(() => {
-    if (addPasskeyOnSignup && !isOpen) {
+    if (addPasskeyOnSignup) {
       openAuthModal();
+
       setAuthStep({
         type: "passkey_create",
       });
     }
-  }, [addPasskeyOnSignup, isOpen, openAuthModal, setAuthStep]);
+  }, [addPasskeyOnSignup, openAuthModal, setAuthStep]);
 
-  useNewUserSignup(handleSignup, isConnected);
```

Comment:
this solves a race condition where an on success handler would set the step to complete and close the modal that was just opened. This makes sure that all handlers complete we're done with the setup flow prior to add passkey

Reply:
yay thanks for looking into this!